Drivers: SPI: Add new  SPI  module to PG23, PG28

silabs-vifisch opened the pull request:
Our goal is to extend usart capabilities to help users to use additional SPI device aout of the box.

signed-off by: silabs-vifisch <viktor.fisch@silabs.com>

jhedberg commented:
@silabs-vifisch you need to fix your git author and signed-off-by information to be of the format `Firstname Lastname <email>`. Also, please just use `git commit -s` to generate the signed-off-by line, since it looks like you've manually created it (as yours starts with a lower-case letter)

jerome-pouiller reviewed (commented):
`silabs` has to be mentioned somewhere in the commit title. Usually, we prefix with `dts: arm: silabs: `

Fix typo in `aout of the box`.

Martinhoff-maker commented on boards/silabs/dev_kits/pg23_pk2504a/pg23_pk2504a-pinctrl.dtsi:
```diff
@@ -21,6 +21,20 @@
 
+	eusart1_default: eusart1_default {
+		group0 {
+			pins = <EUSART1_TX_PC3>, <EUSART1_SCLK_PC1>;
```
I agree that we have alternate function, but since we are on board level, those pin are physically connected to the lcd segment and nothing else, how will you use the eusart then ?

silabs-vifisch replied:
This picture is for pg23 board. The problem with this that all pins are reserved, not connected or may used by LCD if it is enabled, only B4-B6,  are available but for SPI we need 4 pins. So I have not much choice.
Concerning pg28 I reconfigured SPI pin out I've already committed.

asmellby replied:
If the board doesn't provide SPI functionality, there shouldn't be SPI configuration in the board devicetree file.

silabs-vifisch replied:
Actually we can configure SPI if LCD is not used.

jerome-pouiller commented on samples/drivers/spi_bitbang/src/main.c:
You probably didn't want to commit these changes.

jerome-pouiller reviewed (commented):
Issue with PG23 is not yet fixed.

Split commits in an atomic way (for this PR, either one commit for PG28 and one commit for PG23, either just one commit for all changes)

BTW, please check your changes before to send new version of your PR (or convert it to a draft).

silabs-vifisch commented on boards/silabs/dev_kits/pg28_pk2506a/pg28_pk2506a-pinctrl.dtsi:
```diff
@@ -21,6 +21,20 @@
 
+	usart0_default: usart0_default {
+		group0 {
+			pins = <USART0_TX_PD11>, <EUSART1_SCLK_PD13>;
```
eusart1 is is already defined in the .dts as an uart, it is probably used by another module. This is the reason why I decided  to introduce another usart module dedicated to SPI. I looked for available pin and find PD11-PD13 as we agreed above as it is assigned to SPI. So I looked which usart module can be routed to that pins, and found out his was uasrt0.

silabs-vifisch commented on boards/silabs/dev_kits/pg28_pk2506a/pg28_pk2506a.dts:
```diff
@@ -168,6 +168,18 @@
+	#address-cells = <1>;
+	#size-cells = <0>;
+	clock-frequency = <4000000>;
+	pinctrl-0 = <&eusart0_default>;
```
usart0_default is defined as usart0 is used, it compiled this way.

silabs-vifisch committed:
samples: bluetooth: silabs: PG28 dts file changed

 usart0 module is introduced and pg28_pk2506a dts files have been changed
 accordingly.

Signed-off-by: Viktor Fisch <viktor.fisch@silabs.com>

jhedberg commented:
@silabs-vifisch please improve the commit message a bit more. It's not really describing the change properly IMO. It's also good to answer the question "why?" in the commit message, as that may not be clear to everyone.

silabs-vifisch committed:
dts: spi: silabs: usart0 module added to PG28 dts file

 usart0 module is introduced and pg28_pk2506a dts files have been changed
 accordingly. Incorrect pin configuration for EUSART1_SCLK_PD13 heas been
fixed

Signed-off-by: Viktor Fisch <viktor.fisch@silabs.com>
